src/blenderbim/blenderbim/bim/module/search/operator.py

```diff
@@ -381,3 +381,106 @@ def draw(self, context):
         row = self.layout.row(align=True)
         row.operator("bim.toggle_filter_selection", text="Select All").action = "SELECT"
         row.operator("bim.toggle_filter_selection", text="Deselect All").action = "DESELECT"
+
+
+class CreateIfcClassSelSet(bpy.types.Operator):
+    "Click to generate a selection set for all ifc classes"
+    bl_idname = "bim.create_ifcclass_ss"
+    bl_label = "Create IfcClass Selection Set"
+
+    def execute(self, context):
+        if "Selection Sets" in bpy.data.collections:
+            ss = bpy.data.collections["Selection Sets"]  
+        else:
+            ss = bpy.data.collections.new("Selection Sets")
+            ss.color_tag = "COLOR_04"
+
+        if "Ifc Classes" in bpy.data.collections:
+            ifc_classes = bpy.data.collections["Ifc Classes"]
+        else:
+            ifc_classes = bpy.data.collections.new("Ifc Classes")
+            ifc_classes.color_tag = "COLOR_04"
+            ss.children.link(ifc_classes)
+            
+        objects = context.selectable_objects
+        
+        for obj in objects:
+            ifc_element = tool.Ifc.get_entity(obj)
+            if not ifc_element:
+                continue     
+            
+            ifc_type = ifc_element.is_a()
+            if ifc_type in bpy.data.collections:
```

Comment:
I don't know for sure that this will have a beneficial effect on performance, but since we can have thousands of objects and thousands of collections, I assume this lookup can be optimised using a set for instance :
```
ifc_classes_set = set()
for obj in objects:
    ifc_element = tool.Ifc.get_entity(obj)
    if not ifc_element:
        continue     
    ifc_classes_set.add(ifc_element.is_a())
for ifc_class in ifc_classes_set:
    ifc_class_col = bpy.data.collections.get(ifc_type)
    if ifc_class_col is None:
        ifc_class_col = bpy.data.collections.new(f"{ifc_type}")
        ifc_class_col .color_tag = "COLOR_04"
        ifc_classes.children.link(ifc_class_col)
for obj in objects: 
    # Again, avoiding costly lookups ? Might have to test this since try/except can generate overhead too
    try:
        ifc_class.objects.link(obj)
    except RuntimeError:  # Will throw if object already in collection
        pass